azure/converters/rules_test.go

```diff
@@ -0,0 +1,214 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package converters
+
+import (
+	"testing"
+
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2021-02-01/network"
+	"github.com/Azure/go-autorest/autorest/to"
+	. "github.com/onsi/gomega"
+	infrav1 "sigs.k8s.io/cluster-api-provider-azure/api/v1beta1"
+)
+
+func Test_SecurityRuleToSDK(t *testing.T) {
+	cases := []struct {
+		name         string
+		securityRule infrav1.SecurityRule
+		expect       func(*GomegaWithT, network.SecurityRule)
+	}{
+		{
+			name: "convert security role with security group protocol all",
+			securityRule: infrav1.SecurityRule{
+				Name:             "fake-rule",
+				Description:      "fake rule description",
+				Source:           to.StringPtr("fake-source"),
+				SourcePorts:      to.StringPtr("fake-source-ports"),
+				Destination:      to.StringPtr("fake-destination"),
+				DestinationPorts: to.StringPtr("fake-destination-port-ranges"),
+				Priority:         2,
+				Protocol:         infrav1.SecurityGroupProtocolAll,
+				//Direction: infrav1.SecurityRuleDirectionOutbound,
```

Comment:
Remove comment? I think it's fine to not test combinations of `rule.Protocol` and `rule.Direction` here since it'll have way too many combinations.